Dockerfile

```diff
@@ -163,28 +164,50 @@ RUN --mount=type=cache,sharing=shared,id=cargo_registry_index,target=${CARGO_HOM
     # Meanwhile if you want to update wasm-bindgen you also need to update version in:
     #  - packages/wasm-dpp/Cargo.toml
     #  - packages/wasm-dpp/scripts/build-wasm.sh
-    cargo install --profile "$CARGO_BUILD_PROFILE" wasm-bindgen-cli@0.2.86
+    cargo install --profile "$CARGO_BUILD_PROFILE" wasm-bindgen-cli@0.2.86 cargo-chef@0.1.67 --locked
 
 #
-# LOAD SOURCES
+# Rust build planner to speed up builds
 #
-FROM deps as sources
-
+FROM deps AS build-planner
 WORKDIR /platform
-
 COPY . .
+RUN source $HOME/.cargo/env && \
+    cargo chef prepare --recipe-path recipe.json
```

Comment:
COPY . . means we will restart planner every time any file changes, and the cached dependencies will become outdated. Not sure how it works, but maybe it's possible to only copy cargo.toml/lock ? This way planner output could be cached.

Or maybe we can generate it in GitHub actions before calling Docker?

Reply:
This is intended. Planner needs the entire project structure and we just create a recipe. If the recipe is changed then we invalidate build dependencies step. Please check documentation https://github.com/LukeMathWalker/cargo-chef